remote_fetch.go

```diff
@@ -92,13 +92,23 @@ func decompressResponse(bodyBytes []byte, contentLength int64, cacheLocator Cach
 				return errorExtractingPostgres(err)
 			}
 
-			cacheLocation, _ := cacheLocator()
+			// to prevent file corruption when multiple processes attempt to download at the same time we download
+			// first to a cache location, and then move the file into place.
+			tmp, err := os.CreateTemp("", "embedded_postgres")
+			if err != nil {
+				return errorExtractingPostgres(err)
+			}
 
+			if err := os.WriteFile(tmp.Name(), archiveBytes, file.FileHeader.Mode()); err != nil {
+				return errorExtractingPostgres(err)
+			}
+
+			cacheLocation, _ := cacheLocator()
 			if err := os.MkdirAll(filepath.Dir(cacheLocation), 0755); err != nil {
 				return errorExtractingPostgres(err)
 			}
 
-			if err := os.WriteFile(cacheLocation, archiveBytes, file.FileHeader.Mode()); err != nil {
+			if err := os.Rename(tmp.Name(), cacheLocation); err != nil {
```

Comment:
reading through the code for `natefinch/atomic` this should be fine on unix, but may not be enough for windows. I'm not sure if that is a requirement at the moment?